packages/cli/test/integration/publicApi/credentials.test.ts

```diff
@@ -0,0 +1,313 @@
+
+	const authOwnerAgent = utils.createAgent(app, {
+		apiPath: 'public',
+		version: 1,
+		auth: true,
+		user: ownerShell,
+	});
+	const payloads = [credentialPayload(), credentialPayload()];
+	// @ts-ignore
+	delete payloads[1].nodesAccess;
+
+	// calls need to be in sequence for sqlite (cannot run multiple transactions in parallel)
+	const responses = [];
+	for (const payload of payloads) {
+		responses.push(await authOwnerAgent.post('/credentials').send(payload));
+	}
+
+	await Promise.all(
+		responses.map(async (response, index) => {
+			expect(response.statusCode).toBe(200);
+
+			const { id, name, type } = response.body;
+
+			expect(name).toBe(payloads[index].name);
+			expect(type).toBe(payloads[index].type);
+
+			const credential = await Db.collections.Credentials!.findOneOrFail(id);
+
+			expect(credential.name).toBe(payloads[index].name);
+			expect(credential.type).toBe(payloads[index].type);
+			expect(credential.data).not.toBe(payloads[index].data);
+
+			const sharedCredential = await Db.collections.SharedCredentials!.findOneOrFail({
+				relations: ['user', 'credentials'],
+				where: { credentials: credential },
+			});
+
+			expect(sharedCredential.user.id).toBe(ownerShell.id);
+			expect(sharedCredential.credentials.name).toBe(payloads[index].name);
+		}),
+	);
+});
+
+test('POST /credentials should fail with invalid inputs', async () => {
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
+	ownerShell = await testDb.addApiKey(ownerShell);
+
+	const authOwnerAgent = utils.createAgent(app, {
+		apiPath: 'public',
+		version: 1,
+		auth: true,
+		user: ownerShell,
+	});
+
+	await Promise.all(
+		INVALID_PAYLOADS.map(async (invalidPayload) => {
+			const response = await authOwnerAgent.post('/credentials').send(invalidPayload);
+			expect(response.statusCode === 400 || response.statusCode === 415).toBe(true);
+		}),
+	);
+});
+
+test('POST /credentials should fail with missing encryption key', async () => {
+	const mock = jest.spyOn(UserSettings, 'getEncryptionKey');
+	mock.mockRejectedValue(new Error(RESPONSE_ERROR_MESSAGES.NO_ENCRYPTION_KEY));
+
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
+	ownerShell = await testDb.addApiKey(ownerShell);
+
+	const authOwnerAgent = utils.createAgent(app, {
+		apiPath: 'public',
+		version: 1,
+		auth: true,
+		user: ownerShell,
+	});
+
+	const response = await authOwnerAgent.post('/credentials').send(credentialPayload());
+
+	expect(response.statusCode).toBe(500);
+
+	mock.mockRestore();
+});
+
+test('POST /credentials should ignore ID in payload', async () => {
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
+	ownerShell = await testDb.addApiKey(ownerShell);
+
+	const authOwnerAgent = utils.createAgent(app, {
+		apiPath: 'public',
+		version: 1,
+		auth: true,
+		user: ownerShell,
+	});
+
+	const firstResponse = await authOwnerAgent
+		.post('/credentials')
+		.send({ id: '8', ...credentialPayload() });
```

Comment:
We should error if they try to send something that is defined in the spec. We can do that with additionalProperties: false